uber/site_sections/registration.py

```diff
@@ -178,7 +185,12 @@ def qrcode_generator(self, data):
         buffer = BytesIO()
         checkin_barcode.save(buffer, "PNG")
         buffer.seek(0)
-        return cherrypy.lib.file_generator(buffer)
+        png_file_output = cherrypy.lib.file_generator(buffer)
+
+        # set response headers last so that exceptions are displayed properly to the client
+        cherrypy.response.headers['Content-Type'] = "image/png"
```

Comment:
i learned this one through CSV madness, basically if you set this header, then an exception happens, then the exception text doesn't get shown, it comes back as 'unable to display image'